infra/docker-compose/docker-compose.dev.yml

```diff
@@ -0,0 +1,70 @@
+version: "3.7"
+
+services:
+  core:
+    image: maven:3.6-openjdk-11
+    volumes:
+      - ${HOME}/.m2:/root/.m2:delegated
+      - ../../.:/code:cached
+    environment:
+      DB_HOST: db
+      FEAST_STREAM_OPTIONS_BOOTSTRAPSERVERS: kafka:9092
+      GOOGLE_APPLICATION_CREDENTIALS: /etc/gcloud/service-accounts/key.json
+    restart: on-failure
+    depends_on:
+      - db
+      - kafka
+    ports:
+      - 6565:6565
+
+    working_dir: /code
+    command:
+      - mvn
+      - -pl
+      - core
+      - spring-boot:run
```

Comment:
If the aim is to `docker-compose exec core` and do `mvn -pl core spring-boot:run`
to test changes after you make changes in the core's code then maybe we shouldn't 
run core on `docker-compose up`, as the container will stop as soon as `mvn -pl core spring-boot:run` completes.

`sleep 9999d` can suffice here.

Reply:
interesting.  I need to try out some approaches but that is good feedback.

Reply:
swapping out the command for
```
    entrypoint: /bin/sh
    stdin_open: true
    tty: true
```
does the trick.  Then you can run:
`docker-compose -f docker-compose.dev.yml exec core mvn -pl core spring-boot:run` or whatever you want.  I can update the PR, if this is what you prefer.  I like it but it definitely needs some docs for first-time users.  I think a bit of reorganization of directories and files could make it easier to grok and use.

Reply:
@dr3s I think we have different ideas of how this PR should work. Seems like you where going for a more `mvn compile` in host & `docker-compose up` kind of workflow from your discussion. Could you update the PR description with a ideal flow of how this PR should work?

Reply:
sure I will do that.  I'm open to changing it the PR as it stands was expecting:

1. make changes (IDE would likely incremental compile)
2. `docker-compose up` would compile as necessary and run
3. manual testing whatever

The comment above about leaving the container running was in response to your feedback.  I can go that direction.  I just haven't yet built a whole workflow.  This can wait until then but I thought it valuable as-is for first-time users.

Reply:
In that case I think this is fine.